File: parsinghex/ParsingHex.java
import java.io.*;
import java.util.ArrayList;

class ParsingHex {

    private static void parse_line(String s) {
        boolean seen_0 = false;
        boolean in_hex = false;
        String word;
        int start = 0;
        int end = 0;
        int digits = 0;

        for (int i = 0; i < s.length(); i++) {
            if (seen_0 && (s.charAt(i) == 'x' || s.charAt(i) == 'X')) {
                in_hex = true;
            } else if (s.charAt(i) == '0') {
                 seen_0 = true;
                 start = i;
                 end = start;
            } else if (in_hex) {
                while (('a' <= s.charAt(i) && s.charAt(i) <= 'f') || ('A' <= s.charAt(i) && s.charAt(i) <= 'F')) {
                    end++;
                    i++;
                }
                word = Integer.parseInt(s.substring(start, end));
                System.out.printf("0x%X %d\n", word, word);
                start = end;


            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        ArrayList<String> lines = new ArrayList<>();
        String line;
        while ((line = in.readLine()) != null) {
            parse_line(line);
        }

    }

}

File: parsinghex/parsinghex.cpp
#include <bits/stdc++.h>
using namespace std;

void parse_line(string s) {
    bool seen_0 = false;
    bool in_hex = false;
    long int number = 0;
    int digits = 0;

    for (unsigned int i = 0; i < s.length(); i++) {
        if (seen_0 && (s[i] == 'x' || s[i] == 'X')) {
            in_hex = true;
        } else if (s[i] == '0') {
             seen_0 = true;
        } else if (in_hex) {
            while (('a' < s[i] && s[i] < 'f') || ('A' < s[i] && s[i] < 'F')) {
                digits += 1;






int main()
{
    cin.sync_with_stdio(false);
    string s;
    while (getline(cin, s)) {
        parse_line(s);
    }
    return 0;
}

